### ISSUE TITLE
SettingLink not supported?

### ISSUE THREAD

**Issue by MikeManzo** [orchetect/MacControlCenterUI#10]
Xcode 15 (beta)
macOS 14 (beta)

Legacy method for displaying "Settings" is no longer supported:
```swift
NSApp.sendAction(Selector(("showSettingsWindow:")), to: nil, from: nil)
```

The new approach is to use `SettingsLink`.  I am not sure if it's a bug, or if I am using it incorrectly, but using `SettngsLink` with a `MenuCommand` has no effect.

**Comment by orchetect** [comment]
This is a new SwiftUI issue on Xcode 15 / macOS Sonoma.

I don't have Sonoma installed at the moment to test, but I may be able to implement a variation of MenuCommand that conditionally uses `SettingsLink` internally when running Sonoma, and default to the sendAction methods for older macOS versions.

**Comment by orchetect** [comment]
Did some digging and good old mirror reflection gives some hints.

```swift
▽ SwiftUI.SettingsLink<SwiftUI.DefaultSettingsLinkLabel>
  - label: SwiftUI.DefaultSettingsLinkLabel
  ▽ _openSettings: SwiftUI.Environment<SwiftUI.OpenSettingsAction>
    ▽ content: SwiftUI.Environment<SwiftUI.OpenSettingsAction>.Content.keyPath
      - keyPath: \EnvironmentValues.<computed 0x000079283f70b (OpenSettingsAction)> #0
        - super: Swift.KeyPath<SwiftUI.EnvironmentValues, SwiftUI.OpenSettingsAction>
          - super: Swift.PartialKeyPath<SwiftUI.EnvironmentValues>
            ▽ super: Swift.AnyKeyPath
              - _kvcKeyPathStringPtr: nil
```

Apple is using a private environment method called `_openSettings`. Why don't they just make it public so we can use it?

**Comment by MikeManzo** [comment]
Thanks for digging around - I appreciate it.  I was able to 'at least' get SettingsLink into the menu, but it's styled very differently and looks like garbage.  

I have been looking into alternative mens to open the Settings view as a separate window using NSPanel and NSHostingView.  It works ... but the TabView will not show icons, just titles.  It's 'almost' what I can use, but it's not there yet.  Frustrating.

**Comment by orchetect** [comment]
Frustrating indeed. It might be worth submitting a radar to Apple asking for the `openSettings` environment method to be exposed publicly. `SettingsLink` is far too limiting.

Like I mentioned, I have been able to get SettingsLink to look and behave like a `MenuCommand`, however since the `SettingsLink` eats up mouse events like a black hole, I'm still searching for a solution to detect when it's clicked.

**Comment by orchetect** [comment]
Update:

I found a very clever solution for this. Only found it on a hunch. `SettingsLink` contains a SwiftUI `Button` and trying to add custom gestures on it simply does not work. However, by way applying a custom `PrimitiveButtonStyle`, it's possible to call code before and after the `SettingsLink` is clicked by the user.

I will push a new release to the repo once I have it finished and tested.

**Comment by orchetect** [comment]
The *settingslink* branch now has a fully functioning `MenuSettingsCommand` to take the place of `SettingsLink`. It's a special `MenuCommand` variant for opening the Setting scene.

I am now exploring viable options for being able to open the Settings scene programmatically as well. Once I come up with a solution I will merge to main and release.

**Comment by orchetect** [comment]
The solution I've come up with is fairly simple and no direct SettingsLink use needed.

I built a new package called [SettingsAccess](https://github.com/orchetect/SettingsAccess). MacControlCenterUI will remain unchanged. To open Settings, just import SettingsAccess and use its `openSettings()` method in a `MenuCommand`.

I've updated the Demo project in MacControlCenterUI to show it working. In release 2.0.4.
